File: day-12/star-1/main.go
package main

import (
	"io/ioutil"
	"log"
	"strings"
)

func main() {
	// parse input and generate adjacency matrix
	// filepath := "./problem/input.txt"
	// filepath := "./problem/sample.txt"
	filepath := "./problem/custom.txt"
	input, err := ioutil.ReadFile(filepath)
	if err != nil {
		log.Fatal(err)
	}
	vals := parseInput(input)
	adjList := buildAdjacencyList(vals)
	log.Println(adjList)
}

func parseInput(in []byte) [][]uint8 {
	var ret [][]uint8
	rows := strings.Split(string(in), "\n")
	for i := range rows {
		var currRow []byte
		for j := range rows[i] {
			currRow = append(currRow, rows[i][j])
		}
		ret = append(ret, currRow)
	}
	return ret
}

File: day-12/star-1/main.py
def parse(file):
    rows = []
    for row in file.readlines():
        toAdd = []
        for char in row:
            if char != '\n':
                toAdd += char
        rows.append(toAdd)
    return rows

def find_start(input):
    for i in range(len(input)):
        for j in range(len(input[0])):



def main():
    filepath = "./problem/sample.txt"
    file = open(filepath, "r")
    input = parse(file)
    my_list = [1,2,3]


if __name__ == "__main__":
    main()
